src/components/TextInput/BaseTextInput.js

```diff
@@ -168,34 +168,28 @@ function BaseTextInput(props) {
         [props.autoGrowHeight, props.multiline],
     );
 
-    useEffect(() => {
-        // Handle side effects when the value gets changed programatically from the outside
-
-        // In some cases, When the value prop is empty, it is not properly updated on the TextInput due to its uncontrolled nature, thus manually clearing the TextInput.
-        if (inputValue === '') {
-            input.current.clear();
-        }
+    // The ref is needed when the component is uncontrolled and we don't have a value prop
+    const hasValueRef = useRef(initialValue.length > 0);
+    const inputValue = props.value || '';
```

Comment:
what is the difference between `initialValue` and `inputValue` and their purpose? Unable to understand clearly.

Reply:
`initialValue` is the value that we have when the component got created for the first time.
`inputValue` is the value prop but with the fallback to an empty string (instead of undefined)